src/robusta/core/reporting/base.py

```diff
@@ -34,6 +35,25 @@ def __str__(self):
         return f"annotations: {self.annotations} Enrichment: {self.blocks} "
 
 
+class Filterable:
+    @property
+    def attribute_map(self) -> Dict[str, str]:
+        raise NotImplementedError
+
+    def attribute_matches(self, attribute: str, expression: str) -> bool:
+        value = self.attribute_map[attribute]
+        return bool(re.match(expression, value))
+
+    def matches(self, requirements: Dict[str, str]) -> bool:
+        for attribute, expression in requirements.items():
+            if not (
+                attribute in self.attribute_map.keys()
```

Comment:
Should we add a warning log, in case the match attribute is illegal? (in case the user had a typo in the sink definition)
It can be here, or a validation when creating the sink.
WDYT?

Reply:
Sounds like a good idea!

Reply:
Added the logging in the last commit